app/src/main/java/org/fossasia/pslab/others/GPSLogger.java

```diff
@@ -26,18 +28,28 @@
 public class GPSLogger {
 
     public static final int MY_PERMISSIONS_REQUEST_LOCATION = 99;
+    private static final int UPDATE_INTERVAL_IN_MILLISECONDS = 1000;
+    private static final int MIN_DISTANCE_CHANGE_FOR_UPDATES = 1;
     private LocationManager locationManager;
     private Context context;
-    public static boolean permissionAvailable = false;
     private Location bestLocation;
+    private LuxMeterActivity callerActivity;
+    private SharedPreferences sharedPref;
 
     public GPSLogger(Context context, LocationManager locationManager) {
         this.context = context;
         this.locationManager = locationManager;
-        callerActivity = (LuxMeterActivity) context;
+        if(context instanceof LuxMeterActivity){
+            callerActivity = (LuxMeterActivity) context;
+        }
+        sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
     }
 
-    private LuxMeterActivity callerActivity;
+    public GPSLogger(Context context) {
```

Comment:
Move this constructor to top and reformat all the files there are some missing indents and spaces.